src/server/auth/server/api_server.go

```diff
@@ -533,188 +367,61 @@ func (a *apiServer) Deactivate(ctx context.Context, req *auth.DeactivateRequest)
-		req.Principal: roles,
-	}); err != nil {
-		return nil, err
-	}
-	return &auth.ModifyClusterRoleBindingResponse{}, nil
-}
-
-func (a *apiServer) applyClusterRoleBindings(ctx context.Context, roleBindings map[string][]auth.ClusterRole) error {
-	// Check user is authenticated as a super admin for both ModifyClusterRoleBinding and ModifyAdmins
-	// Don't allow users to modify role bindings for the root user. This ensures the root user always has
-	// super admin status, so they can fix the auth config if things are misconfigured.
-	for principal := range roleBindings {
-		if principal == auth.RootUser {
-			return fmt.Errorf("cluster role bindings for %v cannot be modified", auth.RootUser)
-		}
-
-		if err := a.checkCanonicalSubject(principal); err != nil {
-			return errors.Wrapf(err, "subject %q is invalid", principal)
-		}
+		return errors.Wrapf(err, "error confirming Pachyderm Enterprise token")
 	}
 
-	// Update "admins" list (watchAdmins() will update admins cache)
-	if _, err := col.NewSTM(ctx, a.env.GetEtcdClient(), func(stm col.STM) error {
-		admins := a.admins.ReadWrite(stm)
-		fsAdmins := a.fsAdmins.ReadWrite(stm)
-
-		// apply each roleBinding
-		for principal, roles := range roleBindings {
-			// Check if the current modify request contains a SUPER role for this principal
-			var grantSuper bool
-			var grantFS bool
-			for _, role := range roles {
-				if role == auth.ClusterRole_SUPER {
-					grantSuper = true
-				}
-				if role == auth.ClusterRole_FS {
-					grantFS = true
-				}
-			}
-
-			if grantSuper {
-				admins.Put(principal, epsilon)
-			} else {
-				if err := admins.Delete(principal); err != nil && !col.IsErrNotFound(err) {
-					return err
-				}
-			}
-
-			if grantFS {
-				fsAdmins.Put(principal, epsilon)
-			} else if err := fsAdmins.Delete(principal); err != nil && !col.IsErrNotFound(err) {
-				return err
-			}
-		}
-
+	if state == enterpriseclient.State_ACTIVE {
 		return nil
-	}); err != nil {
-		return errors.Wrapf(err, "error applying cluster role bindings")
 	}
-	return nil
-}
 
-// expiredClusterAdminCheck enforces that if the cluster's enterprise token is
-// expired, only admins may log in.
-func (a *apiServer) expiredClusterAdminCheck(ctx context.Context, username string) error {
-	state, err := a.getEnterpriseTokenState()
-	if err != nil {
-		return errors.Wrapf(err, "error confirming Pachyderm Enterprise token")
-	}
-
-	isAdmin, err := a.hasClusterRole(ctx, username, auth.ClusterRole_SUPER)
+	callerInfo, err := a.getAuthenticatedUser(ctx)
 	if err != nil {
 		return err
 	}
-	if state != enterpriseclient.State_ACTIVE && !isAdmin {
-		return errors.New("Pachyderm Enterprise is not active in this " +
-			"cluster (until Pachyderm Enterprise is re-activated or Pachyderm " +
-			"auth is deactivated, only cluster admins can perform any operations)")
+
+	if callerInfo.Subject == auth.RootUser || callerInfo.Subject == auth.PpsUser {
```

Comment:
`expiredClusterAdminCheck` now permits only the PPS user or root user to make calls to a cluster where the enterprise license has run out.

Reply:
**Update** I looked through the existing code and realized that the set of functions that fail if enterprise is deactivated predates this PR and is kind of random:
```
GetACLInTransaction
SetACLInTransaction
SetScopeInTransaction
GetScopeInTransaction
SetACLInTransaction
```

I think ideally it would just be Authenticate(), but changing this seems like it's outside the scope of this PR